python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+    camera : lsst.afw.cameraGeom.Camera
+        Camera object
+    ampName : str
+        amplifier name, eg. C00
+    ampCol : int
+        starting column number for ROI
+    ampRow : int
+        starting row number for ROI
+
+    Returns
+    -------
+    ccdimg : array
+        ccd image array filled with ROI
+    """
+
+    stamp_ccd = amp_to_ccdview(stamp, detector, ampName)
+    ampRows, ampCols = stamp_ccd.shape
+
+    # get corner0 of the location for the ROI
+    lct = LsstCameraTransforms(camera, detector.getName())
+    corner0_CCDX, corner0_CCDY = lct.ampPixelToCcdPixel(ampCol, ampRow, ampName)
+    corner0_CCDX = int(corner0_CCDX)
+    corner0_CCDY = int(corner0_CCDY)
+
+    # get opposite corner, corner2, of the ROI
+    amp = detector[ampName]
+    if amp.getRawFlipX():
+        corner2_CCDX = corner0_CCDX - ampCols
+    else:
+        corner2_CCDX = corner0_CCDX + ampCols
+
+    if amp.getRawFlipY():
+        corner2_CCDY = corner0_CCDY - ampRows
+    else:
+        corner2_CCDY = corner0_CCDY + ampRows
+
+    # now place the ROI
+    try:
+        ccdimg[
+            min(corner0_CCDY, corner2_CCDY) : max(corner0_CCDY, corner2_CCDY),
+            min(corner0_CCDX, corner2_CCDX) : max(corner0_CCDX, corner2_CCDX),
+        ] = stamp_ccd
+    except Exception:
+        print(detector.getName(), ampName)
+        print("Y: ", min(corner0_CCDY, corner2_CCDY), max(corner0_CCDY, corner2_CCDY))
+        print("X: ", min(corner0_CCDX, corner2_CCDX), max(corner0_CCDX, corner2_CCDX))
+
+    return ccdimg
+
+
+def amp_to_ccdview(stamp, detector, ampName):
+    """
+    Comvert a Guider ROI stamp image from Amp view to CCD view
+
+    Parameters
+    ----------
+    stamp : array
+        raw ROI array in amp coordinates
+    detector : lsst.afw.cameraGeom.Detector
+        Detector of interest.
+    ampName : str
+        amplifier name, eg. C00
+
+    Returns
+    -------
+    img : array
+        ROI image flipped to be in CCD coordinate orientation
+    """
+    amp = detector[ampName]
+    img = stamp.copy()
+    if amp.getRawFlipX():
+        img = np.fliplr(img)
+    if amp.getRawFlipY():
+        img = np.flipud(img)
+    return img
+
+
+def convert_to_focal_plane(xccd: np.ndarray, yccd: np.ndarray, detNum: int) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert from ccd pixels coordinates to focal plane coordinates.
+
+    Parameters
+    ----------
+    xccd : np.ndarray
+        Array of x CCD pixel coordinates.
+    yccd : np.ndarray
+        Array of y CCD pixel coordinates.
+    detNum : int
+        Detector number (e.g., 189 for R22_S11).
+
+    Returns
+    -------
+    xfp, yfp : np.ndarray
+        Arrays of x and y focal plane coordinates (mm).
+
+    """
+    if len(xccd) > 0:
+        detector = LsstCam.getCamera()[detNum]
+        # Convert the star positions to focal plane coordinates
+        xfp, yfp = pixel_to_focal(xccd, yccd, detector)
+    else:
+        xfp, yfp = np.array([]), np.array([])
```

Comment:
curious what the purpose is of these empty conversions here and in the function below - are these actually necessary?

Reply:
Make mypy happy